airflow/providers/amazon/aws/operators/batch.py

```diff
@@ -108,12 +110,13 @@ class BatchOperator(BaseOperator):
         "job_queue",
         "overrides",
```

Comment:
TIL the names here need to match not the parameter name in the ctor but the name of the attribute in the class, so:
```suggestion
        "container_overrides",
```

Thinking about it, I wonder if this is a breaking change... I don't know too well how this works 😬

Reply:
@camilleanne @vandonr-amz I recommend deprecate old parameter but keep it for a while, so users would have a time for change their code.
It is much easier achieve in subclasss of `BaseOperator` (this PR case), because all arguments are keyword. 

Some examples

**Use old attribute value only if new attribute not set**
https://github.com/apache/airflow/blob/11f30a887c77f9636e88e31dffd969056132ae8c/airflow/providers/slack/operators/slack_webhook.py#L95-L104

**Use old attribute value only if it equal new attribute value or not new attribute not set**
https://github.com/apache/airflow/blob/11f30a887c77f9636e88e31dffd969056132ae8c/airflow/providers/amazon/aws/operators/athena.py#L91-L101

Reply:
given the fact that it was just a rename for readability, I wonder if it wouldn't be simpler to just keep the old name ?

Reply:
I thought it fine if rename this argument because one day we might also add `eks_properties_overrides`, so `override` might confuse users more rather than change attribute name.
And deprecation warning give time to change arguments in end users DAG code.

Reply:
> TIL the names here need to match not the parameter name in the ctor but the name of the attribute in the class

To translate that into ELI5: if you have `self.foo = bar` in your operator, the template field would be named "foo" to match "self.foo", not "bar". :P